[react/jsx-one-expression-per-line] moves text to the beginning of line

Hi! I tried to use "react/jsx-one-expression-per-line" rule, but its behaviour is very strange and I don't know how can I fix it.

Sandbox : https://codesandbox.io/s/gatsby-starter-default-f7u3w , please do `npm run lint` in terminal

My config `.eslintrc.json`
```json
{
   "parserOptions": {
        "ecmaVersion": 10,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "plugins": [
        "react"
    ],
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended"
    ],
    "rules": {
        "react/jsx-one-expression-per-line": [2, { "allow": "none" }]
    }
}
```

Before running `eslint --fix`
```html
    <h1>Hi people</h1>
    <p>Welcome to your new Gatsby site.</p>
    <p>Now go build something great.</p>
```
After applying eslint
```html
    <h1>
Hi people
</h1>
    <p>
Welcome to your new Gatsby site.
</p>
    <p>
Now go build something great.
</p>
```

Hi @ljharb, I used @lkazberova code as a testing example.
I recreated the same scenario initially mentioned June 23, 2019, and  I did not come across the same issue.
Below is the proper output of eslint with appropriate rules.

rules:
```
    "rules": {
        "react/jsx-one-expression-per-line": [2, { "allow": "none" }],
        "react/jsx-indent": [2, 2]
    },
```
Sample Code Below:
Before running `npx eslint index.js --fix`
```
  <Layout>
    <SEO title="Home" />
    <h1>{"Hi people"}<button/></h1>
    <p>Welcome to your new Gatsby site.</p>
    <p>Now go build something great.</p>
    <h1>Hi people<button/></h1>
    <div style={{ maxWidth: `300px`, marginBottom: `1.45rem` }}>
      <Image />
    </div><Link to="/page-2/">Go to page 2</Link>
  </Layout>
```
Errors found from Eslint according to rules `npx eslint index.js`
```
  11:9   error  `{"Hi people"}` must be placed on a new line                     react/jsx-one-expression-per-line
  11:22  error  `button` must be placed on a new line                            react/jsx-one-expression-per-line
  12:8   error  `Welcome to your new Gatsby site.` must be placed on a new line  react/jsx-one-expression-per-line
  13:8   error  `Now go build something great.` must be placed on a new line     react/jsx-one-expression-per-line
  14:9   error  `Hi people` must be placed on a new line                         react/jsx-one-expression-per-line
  14:18  error  `button` must be placed on a new line                            react/jsx-one-expression-per-line
  17:11  error  `Link` must be placed on a new line                              react/jsx-one-expression-per-line
  17:31  error  `Go to page 2` must be placed on a new line                      react/jsx-one-expression-per-line
```
After applying `npx eslint index.js --fix`
```
  <Layout>
    <SEO title="Home" />
    <h1>
      {"Hi people"}
      <button/>
    </h1>
    <p>
      Welcome to your new Gatsby site.
    </p>
    <p>
      Now go build something great.
    </p>
    <h1>
      Hi people
      <button/>
    </h1>
    <div style={{ maxWidth: `300px`, marginBottom: `1.45rem` }}>
      <Image />
    </div>
    <Link to="/page-2/">
      Go to page 2
    </Link>
  </Layout>
```
Is there still a need to continue to look into this issue?
If so, please let me know.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,10 +14,12 @@ This change log adheres to standards from [Keep a CHANGELOG](https://keepachange
 
 ### Changed
 * [Refactor] [`jsx-indent-props`]: improved readability of the checkNodesIndent function ([#3315][] @caroline223)
+* [Tests] [`jsx-indent`], [`jsx-one-expression-per-line`]: add passing test cases ([#3314][] @ROSSROSALES)
 
 [#3320]: https://github.com/jsx-eslint/eslint-plugin-react/pull/3320
 [#3317]: https://github.com/jsx-eslint/eslint-plugin-react/pull/3317
 [#3315]: https://github.com/jsx-eslint/eslint-plugin-react/pull/3315
+[#3314]: https://github.com/jsx-eslint/eslint-plugin-react/pull/3314
 [#3311]: https://github.com/jsx-eslint/eslint-plugin-react/pull/3311
 
 ## [7.30.1] - 2022.06.23

--- a/tests/lib/rules/jsx-indent.js
+++ b/tests/lib/rules/jsx-indent.js
@@ -45,7 +45,7 @@ ruleTester.run('jsx-indent', rule, {
     {
       code: `
         <App>
-        </App>
+        </App> 
       `,
     },
     {
@@ -2998,6 +2998,135 @@ const Component = () => (
         {
           message: 'Expected indentation of 10 space characters but found 12.',
           line: 6,
+        },
+      ],
+    },
+    {
+      code: `
+        const IndexPage = () => (
+          <h1>
+        {"Hi people"}
+        <button/>
+        </h1>
+        );
+      `,
+      output: `
+        const IndexPage = () => (
+          <h1>
+            {"Hi people"}
+            <button/>
+          </h1>
+        );
+      `,
+      options: [2],
+      errors: [
+        {
+          messageId: 'wrongIndent',
+          data: {
+            needed: 12,
+            type: 'space',
+            characters: 'characters',
+            gotten: 8,
+          },
+        },
+        {
+          messageId: 'wrongIndent',
+          data: {
+            needed: 12,
+            type: 'space',
+            characters: 'characters',
+            gotten: 8,
+          },
+        },
+        {
+          messageId: 'wrongIndent',
+          data: {
+            needed: 10,
+            type: 'space',
+            characters: 'characters',
+            gotten: 8,
+          },
+        },
+      ],
+    },
+    // Would be nice to handle in one pass, but multipass works fine.
+    {
+      code: `
+        const IndexPage = () => (
+          <h1>
+        Hi people
+        <button/>
+        </h1>
+        );
+      `,
+
+      output: `
+        const IndexPage = () => (
+          <h1>
+            Hi people
+        <button/>
+          </h1>
+        );
+      `,
+      options: [2],
+      errors: [
+        {
+          messageId: 'wrongIndent',
+          data: {
+            needed: 12,
+            type: 'space',
+            characters: 'characters',
+            gotten: 8,
+          },
+        },
+        {
+          messageId: 'wrongIndent',
+          data: {
+            needed: 12,
+            type: 'space',
+            characters: 'characters',
+            gotten: 8,
+          },
+        },
+        {
+          messageId: 'wrongIndent',
+          data: {
+            needed: 10,
+            type: 'space',
+            characters: 'characters',
+            gotten: 8,
+          },
+        },
+      ],
+    },
+    {
+      code: `
+        const IndexPage = () => (
+          <h1>
+            Hi people
+        <button/>
+          </h1>
+        );
+      `,
+
+      output: `
+        const IndexPage = () => (
+          <h1>
+            Hi people
+            <button/>
+          </h1>
+        );
+      `,
+      options: [2],
+      errors: [
+        {
+          messageId: 'wrongIndent',
+          data: {
+            needed: 12,
+            type: 'space',
+            characters: 'characters',
+            gotten: 8,
+          },
         },
       ],
     },

--- a/tests/lib/rules/jsx-one-expression-per-line.js
+++ b/tests/lib/rules/jsx-one-expression-per-line.js
@@ -1389,5 +1389,185 @@ a
       ],
       parserOptions,
     },
+    {
+    // TODO: handle in a single pass
+      code: `
+        const IndexPage = () => (
+          <h1>{"Hi people"}<button/></h1>
+        );
+      `,
+      output: `
+        const IndexPage = () => (
+          <h1>
+{"Hi people"}<button/></h1>
+        );
+      `,
+      errors: [
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: '{"Hi people"}' },
+        },
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'button' },
+        },
+      ],
+      parserOptions,
+    },
+    {
+      code: `
+        const IndexPage = () => (
+          <h1>
+{"Hi people"}<button/></h1>
+        );
+      `,
+      output: `
+        const IndexPage = () => (
+          <h1>
+{"Hi people"}
+<button/>
+</h1>
+        );
+      `,
+      errors: [
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'button' },
+        },
+      ],
+      parserOptions,
+    },
+    // TODO: handle in a single pass (see above)
+    {
+      code: `
+        <Layout>
+        <p>Welcome to your new Gatsby site.</p>
+        <p>Now go build something great.</p>
+        <h1>Hi people<button/></h1>
+        </Layout>
+      `,
+      output: `
+        <Layout>
+        <p>
+Welcome to your new Gatsby site.
+</p>
+        <p>
+Now go build something great.
+</p>
+        <h1>
+Hi people<button/></h1>
+        </Layout>
+      `,
+      errors: [
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'Welcome to your new Gatsby site.' },
+        },
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'Now go build something great.' },
+        },
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'Hi people' },
+        },
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'button' },
+        },
+      ],
+      parserOptions,
+    },
+    {
+      code: `
+        <Layout>
+        <p>
+Welcome to your new Gatsby site.
+</p>
+        <p>
+Now go build something great.
+</p>
+        <h1>
+Hi people<button/></h1>
+        </Layout>
+      `,
+      output: `
+        <Layout>
+        <p>
+Welcome to your new Gatsby site.
+</p>
+        <p>
+Now go build something great.
+</p>
+        <h1>
+Hi people
+<button/>
+</h1>
+        </Layout>
+      `,
+      errors: [
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'button' },
+        },
+      ],
+      parserOptions,
+    },
+    // TODO: handle in a single pass
+    {
+      code: `
+        <Layout>
+          <div style={{ maxWidth: \`300px\`, marginBottom: \`1.45rem\` }}>
+            <Image />
+          </div><Link to="/page-2/">Go to page 2</Link>
+        </Layout>
+      `,
+      output: `
+        <Layout>
+          <div style={{ maxWidth: \`300px\`, marginBottom: \`1.45rem\` }}>
+            <Image />
+          </div>
+<Link to="/page-2/">Go to page 2</Link>
+        </Layout>
+      `,
+      errors: [
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'Link' },
+        },
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'Go to page 2' },
+        },
+      ],
+      parserOptions,
+    },
+    {
+      code: `
+        <Layout>
+          <div style={{ maxWidth: \`300px\`, marginBottom: \`1.45rem\` }}>
+            <Image />
+          </div>
+<Link to="/page-2/">Go to page 2</Link>
+        </Layout>
+      `,
+      output: `
+        <Layout>
+          <div style={{ maxWidth: \`300px\`, marginBottom: \`1.45rem\` }}>
+            <Image />
+          </div>
+<Link to="/page-2/">
+Go to page 2
+</Link>
+        </Layout>
+      `,
+      errors: [
+        {
+          messageId: 'moveToNewLine',
+          data: { descriptor: 'Go to page 2' },
+        },
+      ],
+      parserOptions,
+    },
   ]),
 });

